java-checks/src/main/java/org/sonar/java/checks/CombineCatchCheck.java

```diff
@@ -47,23 +51,41 @@ public void visitNode(Tree tree) {
     for (CatchTree catchTree : ((TryStatementTree) tree).catches()) {
       for (CatchTree catchTreeToBeCompared : catches) {
         if (SyntacticEquivalence.areSemanticallyEquivalent(catchTree.block().body(), catchTreeToBeCompared.block().body())) {
-          reportIssue(catchTree, catchTreeToBeCompared);
+          reportIssueWithQuickFix(catchTree, catchTreeToBeCompared);
           break;
         }
       }
       catches.add(catchTree);
     }
   }
 
-  private void reportIssue(CatchTree catchTree, CatchTree catchTreeToBeCompared) {
-    String message = "Combine this catch with the one at line " + LineUtils.startLine(catchTreeToBeCompared.catchKeyword())
-      + ", which has the same body." + context.getJavaVersion().java7CompatibilityMessage();
+  private void reportIssueWithQuickFix(CatchTree catchTree, CatchTree catchTreeToBeCompared) {
+    String quickFixMessage = "Combine this catch with the one at line " + LineUtils.startLine(catchTreeToBeCompared.catchKeyword());
+    String issueMessage = quickFixMessage + ", which has the same body." + context.getJavaVersion().java7CompatibilityMessage();
     List<JavaFileScannerContext.Location> flow = Collections.singletonList(new JavaFileScannerContext.Location("Combine with this catch", catchTreeToBeCompared));
-    reportIssue(catchTree.parameter(), message, flow, null);
+    QuickFixHelper.newIssue(context)
+      .forRule(this)
+      .onTree(catchTree.parameter())
+      .withMessage(issueMessage)
+      .withSecondaries(flow)
+      .withQuickFix( () -> computeQuickFix(catchTree, catchTreeToBeCompared, quickFixMessage) )
+      .report();
   }
 
   @Override
   public boolean isCompatibleWithJavaVersion(JavaVersion version) {
     return version.isJava7Compatible();
   }
+
+  private JavaQuickFix computeQuickFix(CatchTree catchTree, CatchTree catchTreeToBeCompared, String qfMessage) {
+    var builder = JavaQuickFix.newQuickFix(qfMessage);
+    builder.addTextEdit(JavaTextEdit.removeTree(catchTree));
+    SyntaxToken openParent = catchTreeToBeCompared.openParenToken();
+    builder.addTextEdit(
+      JavaTextEdit.insertAfterTree(
+        openParent,
+        QuickFixHelper.contentForTree(catchTree.parameter().type(), context) + " | "));
+    return builder.build();
```

Comment:
We need to check deeper what kind of type is handled by the catch clause as some combinations lead to compilation breaking.
```java
 void writeStuff(String message) {
    try (OutputStream out = new FileOutputStream("/tmp/out.txt")) {
      out.write(message.getBytes(StandardCharsets.UTF_8));
    } catch (FileNotFoundException e) {
      throw new RuntimeException(e);
    } catch (IOException e) {
      throw new RuntimeException(e);
    }
  }
```
In this case, because `FileNotFoundException` is a sub type of `IOException`, the masking breaks compilation